src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelDiscoveryEventsHandler.cs

```diff
@@ -65,84 +59,77 @@ internal class ParallelDiscoveryEventsHandler : ITestDiscoveryEventsHandler2
     /// <inheritdoc/>
     public void HandleDiscoveryComplete(DiscoveryCompleteEventArgs discoveryCompleteEventArgs, IEnumerable<TestCase> lastChunk)
     {
-        var totalTests = discoveryCompleteEventArgs.TotalCount;
-        var isAborted = discoveryCompleteEventArgs.IsAborted;
-
-        // Aggregate for final discovery complete
-        _discoveryDataAggregator.Aggregate(totalTests, isAborted);
-
-        // Aggregate Discovery Data Metrics
-        _discoveryDataAggregator.AggregateDiscoveryDataMetrics(discoveryCompleteEventArgs.Metrics);
+        // Aggregate data for final discovery complete
+        _discoveryDataAggregator.Aggregate(discoveryCompleteEventArgs);
 
         // we get discovery complete events from each host process
         // so we cannot "complete" the actual operation until all sources are consumed
         // We should not block last chunk results while we aggregate overall discovery data
```

Comment:
```suggestion
        // We get DiscoveryComplete events from each ProxyDiscoveryManager (each testhost)
        // and they contain the last chunk of tests that were discovered in that particular testhost.
        // We want to send them to the IDE, but we don't want to tell it that discovery finished, because
        // other sources are still being discovered. So we translate the last chunk to a normal TestsDiscovered
        // event and send that to the IDE. Once all sources are completed, we will send a single DiscoveryComplete.
```